internal/metrics/posthog.go

```diff
@@ -0,0 +1,41 @@
+package metrics
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+
+	"github.com/SwissDataScienceCenter/renku-gateway/internal/config"
+	"github.com/posthog/posthog-go"
+)
+
+type PosthogMetricsClient struct {
+	posthogClient posthog.Client
+}
+
+func (p *PosthogMetricsClient) anonymizeUser(userId string) string {
+	hash := md5.Sum([]byte(userId))
+	return hex.EncodeToString(hash[:])
+}
+
+func (p *PosthogMetricsClient) UserLoggedIn(userId string) error {
+	return p.posthogClient.Enqueue(posthog.Capture{DistinctId: p.anonymizeUser(userId), Event: "user_logged_in"})
+}
+
+func (p *PosthogMetricsClient) Close() {
+	p.posthogClient.Close()
+}
+
+func NewPosthogClient(c config.PosthogConfig) (*PosthogMetricsClient, error) {
```

Comment:
1. When `config.Enabled` is false, this method should either error out or return `nil, nil`.
2. nitpick: if this returns `*PosthogMetricsClient = nil` when `config.Enabled` is false, then the logic in `cmd/gateway/main.go` can be simplified by always calling `login.WithMetricsClient()` (setting a `nil` client when not activated).

Reply:
not resolved